### PR TITLE
cleanup device type override code

### PR DESCRIPTION
**Pull request by murrant**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/common.php [hunk 1]
```diff
@@ -994,11 +994,11 @@ function load_os(&$device)
 
     // Set type to a predefined type for the OS if it's not already set
     $loaded_os_type = Config::get("os.{$device['os']}.type");
-    if ((! isset($device['attribs']['override_device_type']) && $device['attribs']['override_device_type'] != 1) && array_key_exists('type', $device) && $loaded_os_type != $device['type']) {
```
Comment (Jellyfrog) [inline includes/common.php:997]: what about this `!= 1` check?

Comment (murrant) [reply includes/common.php:997]: `! $model->getAttrib('override_device_type')`  is the same.

File: includes/common.php [hunk 2]
```diff
@@ -994,11 +994,11 @@ function load_os(&$device)
...
-        log_event('Device type changed ' . $device['type'] . ' => ' . $loaded_os_type, $device, 'system', 3);
```
Comment (SourceDoctor) [inline includes/common.php:998]: as i remember will log event be printed in Eventlog, why did you remove this?
A device type change is a noteable information for user, in my eyes :-)

**Review by Jellyfrog — APPROVED** [review]

File: includes/common.php (hunk 2 above)
Comment (Jellyfrog) [reply includes/common.php:998]: it's moved to the device observer
